opennms-dao/src/main/java/org/opennms/netmgt/dao/hibernate/TopologyEntityCacheImpl.java

```diff
@@ -0,0 +1,109 @@
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.dao.hibernate;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.opennms.netmgt.dao.api.TopologyEntityCache;
+import org.opennms.netmgt.dao.api.TopologyEntityDao;
+import org.opennms.netmgt.model.CdpLinkTopologyEntity;
+import org.opennms.netmgt.model.NodeTopologyEntity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+
+public class TopologyEntityCacheImpl implements TopologyEntityCache {
+    
+    private final static Logger LOG = LoggerFactory.getLogger(TopologyEntityCacheImpl.class);
+    private final static String KEY = "KEY";
+    private final static String SYSTEM_PROPERTY_CACHE_DURATION = "org.opennms.ui.topology-entity-cache-duration";
+
+    private TopologyEntityDao topologyEntityDao;
+
+    LoadingCache<String, List<NodeTopologyEntity>> nodeTopologyEntities = createCache(
+            new CacheLoader<String, List<NodeTopologyEntity>>() {
+                @Override
+                public List<NodeTopologyEntity> load(String key) {
+                    return topologyEntityDao.getNodeTopologyEntities();
+                }
+            }
+    );
+
+    LoadingCache<String, List<CdpLinkTopologyEntity>> cdpLinkTopologyEntities = createCache(
+            new CacheLoader<String, List<CdpLinkTopologyEntity>>() {
+                @Override
+                public List<CdpLinkTopologyEntity> load(String key) {
+                    return topologyEntityDao.getCdpLinkTopologyEntities();
+                }
+            }
+    );
+
+    private <Key, Value> LoadingCache<Key, Value> createCache(CacheLoader<Key, Value> loader) {
+        return CacheBuilder
+                .newBuilder()
+                .expireAfterWrite(getCacheDuration(), TimeUnit.SECONDS)
+                .build(loader);
+    }
+
+
+    public List<NodeTopologyEntity> getNodeTopolgyEntities() {
+        return this.nodeTopologyEntities.getUnchecked(KEY);
+    }
+
+    public List<CdpLinkTopologyEntity> getCdpLinkTopologyEntities() {
+        return this.cdpLinkTopologyEntities.getUnchecked(KEY);
+    }
+
+    @Override
+    public void refresh(){
+        nodeTopologyEntities.refresh(KEY);
+        cdpLinkTopologyEntities.refresh(KEY);
+    }
+
+    private int getCacheDuration(){
+        String duration = System.getProperty(SYSTEM_PROPERTY_CACHE_DURATION, "300");
```

Comment:
Can use `Integer.getInteger` to retrieve an integer value from the sys prop.

Reply:
You are right, but I believe with Integer.getInteger() we can't distinguish between a not set property and an invalid value (not parsable number):
`propname=123 # will result in 123`
`# propname=123 # will result in default value`
`propname=not_a_number # will result in default value`
2 and 3 will have the same outcome (default value) but we won't have a warning in the log. If thats ok then we can simplify the code by just using Integer.getInteger().
Another option would be to generalize this behavior so we don't have to repeat it everywhere, something like this:
```
SystemHelper.getAsInteger(String name, int defaultValue){
    try {
        return Integer.parseInt(name);
    } catch(NumberFormatException e) {
        LOG.warn(...);
    }
}
```
@j-white What do you think? Is there a general way how to deal with invalid properties?

Reply:
We already use `Integer.getInteger` in a number of places so I'd would just continue to use it here.

You raise a good point though. We could add a utility function like the one you proposed, migrate the existing calls to `Integer.getInteger` to use it, and add a checkstyle rule to prevent new uses of `Integer.getInteger`. Maybe we can tackle this in a separate issue?

Reply:
@j-white, makes sense to me. I changed the code to use Integer.getInteger()
about the separate issue: I created a ticket: https://issues.opennms.org/browse/NMS-10477. Not sure what labels I need to set. I can tackle the issue in the next days if you want, it shoudn't take long.